src/features/navigation/SearchStackNavigator/SearchStackNavigator.tsx

```diff
@@ -1,20 +1,41 @@
 import React from 'react'
 
-import { SearchScreens } from 'features/navigation/SearchStackNavigator/routes'
 import { SEARCH_STACK_NAVIGATOR_SCREEN_OPTIONS } from 'features/navigation/SearchStackNavigator/searchStackNavigationOptions'
 import { SearchStack } from 'features/navigation/SearchStackNavigator/Stack'
-import { SearchStackScreenNames } from 'features/navigation/SearchStackNavigator/types'
+import { SearchStackRoute } from 'features/navigation/SearchStackNavigator/types'
+import { SearchFilter } from 'features/search/pages/SearchFilter/SearchFilter'
+import { SearchLanding } from 'features/search/pages/SearchLanding/SearchLanding'
+import { SearchResults } from 'features/search/pages/SearchResults/SearchResults'
+import { ThematicSearch } from 'features/search/pages/ThematicSearch/ThematicSearch'
+import { SearchView } from 'features/search/types'
 
-export const SearchStackNavigator = ({
-  initialRouteName,
-}: {
-  initialRouteName: SearchStackScreenNames
-}) => {
+export const SearchStackNavigator = () => {
   return (
     <SearchStack.Navigator
-      initialRouteName={initialRouteName}
+      initialRouteName={SearchView.Landing}
       screenOptions={SEARCH_STACK_NAVIGATOR_SCREEN_OPTIONS}>
-      {SearchScreens}
+      {routes.map(({ name, component }) => (
+        <SearchStack.Screen name={name} key={name} component={component} />
```

Comment:
Après discussion avec Bruno E. je vois mieux la logique qu'on met en place, plutôt que d'avoir la définition des routes en dessous, je préfèrerai qu'elles soit ici.
```suggestion
        <SearchStack.Screen name={SearchView.Landing} key={SearchView.Landing} component={SearchLanding} />
        <SearchStack.Screen name={SearchView.Results} key={SearchView.Results} component={SearchResults} />
        etc ...
```
Ce sera sans doute moins compliqué à comprendre si on est déclaratif ici de toutes les routes qui sont dans la stack

Reply:
Oui c'est mieux, merci du retour!